### ISSUE TITLE
make plot functions compatible with VegaLite 2.0.0 

### ISSUE THREAD

**Issue by lrennels** [mimiframework/Mimi.jl#673]
@davidanthoff hopefully this is easy I'll ask you if it gets confusing!

**Comment by davidanthoff** [comment]
I don't think we have to do anything, hopefully. I only bumped it to VegaLite 2.0.0 because it ships with vega-lite 4 instead of 3, which has some breaking changes in some obscure corner cases that I'm 99% sure do not affect us.

**Comment by lrennels** [comment]
I think it’s breaking the plot function, at least it does on CI and locally on my machine dependent on the VegaLite version.  We were kind of hacking it a little but so maybe I was stepping outside the API.

**Comment by lrennels** [comment]
Or maybe I’m crazy!

**Comment by lrennels** [comment]
Our `plot` function looks like this:
```

"""
    plot(m::Model, comp_name::Symbol, datum_name::Symbol; interactive::Bool = false)

Plot a specific `datum_name` (a `variable` or `parameter`) of Model `m`. If the 
Bool `interactive` option is set to `true`, the plot will be interactive which will 
limit saving options.
"""
function plot(m::Model, comp_name::Symbol, datum_name::Symbol; interactive::Bool = false)

    if m.mi === nothing
        error("A model must be run before it can be plotted")
    end
    
    spec = Mimi._spec_for_item(m, comp_name, datum_name, interactive=interactive)
    spec === nothing ? error("Spec cannot be built.") : VLspec = spec["VLspec"]

    return VegaLite.VLSpec{:plot}(VLspec)

end
```
The last line, `VegaLite.VLSpec{:plot}(VLspec)` throws the following error in VegaLite 2.0.0
```

julia> VegaLite.VLSpec{:plot}(VLspec)
ERROR: TypeError: in Type{...} expression, expected UnionAll, got Type{VegaLite.VLSpec}
Stacktrace:
 [1] top-level scope at REPL[92]:1
```

**Comment by lrennels** [comment]
oh ... I think we can just take out the `{:plot}` type specification, but I'm not sure why we needed it before?

**Comment by lrennels** [comment]
fixed it #674
